lib/ansible/modules/cloud/vmware/vmware_guest_disk.py

```diff
@@ -572,8 +572,9 @@ def sanitize_disk_inputs(self):
                 else:
                     current_disk['state'] = disk['state']
 
-            if current_disk['state'] == 'absent':
-                current_disk['destroy'] = disk['destroy']
+            # Reset current_disk['destroy'] to False only when user set not remove disk file explicitly
+            if current_disk['state'] == 'absent' and 'destroy' in disk and not disk['destroy']:
```

Comment:
I think you can just do:

```python
if current_disk['state'] == 'absent':
    current_disk['destroy'] = disk.get('destroy', False)
```

Reply:
Yes, your change also can work.  Thanks, I will modify according to it.